src/azure_devtools/scenario_tests/base.py

```diff
@@ -122,8 +122,12 @@ def __init__(self,  # pylint: disable=too-many-arguments
             recording_dir,
             '{}.yaml'.format(recording_name or method_name)
         )
-        if self.is_live and os.path.exists(self.recording_file):
-            os.remove(self.recording_file)
+        self.temp_recording_file = os.path.join(
+            recording_dir,
+            '{}.temp.yaml'.format(recording_name or method_name)
```

Comment:
your temp file still in recording  directory and when there is a failure, these temp files are still in recording firctory.
Please clean up these temp files in recording folder or use temp folder to store temp files instead of recording folder

Reply:
Recommend to add test for both successful and failure scenarios.